packages/signer/src/api/ws.sagas.ts

```diff
@@ -146,15 +154,41 @@ export function* handleRequest({ socket, request, data }: WebSocketMessage) {
 
   const { signature, publicKey, ...jsonRpcRequest } = fullRequest;
 
-  const permissions: Permission[] = yield select(getPermissions);
-  const origin = request.headers.origin && new URL(request.headers.origin).host;
+  const origin = getOriginHost(request);
+  // Dont serve requests without origin
+  if (origin === undefined || origin === null) {
```

Comment:
```suggestion
  if (origin == null) {
```
or just
```suggestion
  if (!origin) {
```

Reply:
I would prefer being explicit in this case if it is fine with you.

Reply:
Then using the first option should be fine, right?